zephyrproject-rtos/twister
# Align quarantine with TwisterV1

Changed implementation of the quarantine feature to be aligned with Twister V1.
Instead of adding separate plugin to filter quarantined test configurations (there was also an issue with xdist), filtering is done in `should_be_skipped` method. This implementation is closer to TwiterV1.

To test one can use examples from:
https://github.com/gchwier/twister/tree/platform_scope_tests/tests/external
```
python compare_platform_scope.py commands_to_compare.yml --log-level debug --filter-test='bluetooth_with_quarantine.*'
```
or just run TwisterV1 and TwisterV2 and compare results:
```
scripts/twister -T tests/bluetooth --enable-slow --dry-run --quarantine-list quarantine.yml --quarantine-verify
pytest tests/bluetooth --enable-slow --collect-only --quarantine-list=quarantine.yml --quarantine-verify --testplan-json=twister-out/testplan.json
```
